pkgs/games/anki/default.nix

```diff
@@ -1,198 +1,381 @@
+
+    nativeBuildInputs = [
+      fixup_yarn_lock
+      yarn
+      nodejs-slim
     ];
-    sha256 = sha256-pkg;
+
+    configurePhase = ''
+      export HOME=$NIX_BUILD_TOP
+      yarn config --offline set yarn-offline-mirror $yarnOfflineCache
+      fixup_yarn_lock yarn.lock
+      yarn install --offline --frozen-lockfile --ignore-scripts --no-progress --non-interactive
+      patchShebangs node_modules/
+    '';
+
+    installPhase = ''
+      mv node_modules $out
+    '';
   };
 
+  # Mask ninja's setup hooks, we just want the binary, not the implicit 'ninja install'
+  ninja-bin = buildEnv {
+    name = ninja.name;
+    paths = [ ninja ];
+    pathsToLink = [ "/bin" ];
+  };
+in
+python39.pkgs.buildPythonApplication {
+  inherit pname version src;
+
   outputs = [ "out" "doc" "man" ];
 
-  propagatedBuildInputs = [
-    pyqtwebengine
-    sqlalchemy
+  patches = [
+    ./gl-fixup.patch
+    ./no-update-check.patch
+  ];
+
+  buildInputs = [
+    anki-rsbridge
+    qt6.qtbase
+    python39.pkgs.protobuf
+  ];
+  nativeBuildInputs = [
+    fakeCargo
+    fakeCorepack
+    fakeGit
+    fixup_yarn_lock
+    offlineYarn
+
+    ninja-bin
+    python39.pkgs.protobuf
+    qt6.wrapQtAppsHook
+    rsync
+  ];
+  propagatedBuildInputs = with python39.pkgs; [
+    # This rather long list came from running:
+    #    grep --no-filename -oE "^[^ =]*" python/{requirements.base.txt,requirements.bundle.txt,requirements.qt6_4.txt} | \
+    #      sort | uniq | grep -v "^#$"
+    # in their repo at the git tag for this version
+    # There's probably a more elegant way, but the above extracted all the
+    # names, without version numbers, of their python dependencies. The hope is
+    # that nixpkgs versions are "close enough"
+    attrs
     beautifulsoup4
-    send2trash
-    pyaudio
-    requests
+    build
+    certifi
+    charset-normalizer
+    click
+    colorama
     decorator
-    markdown
+    distro
+    flask
+    flask-cors
+    idna
+    importlib-metadata
+    itsdangerous
+    jinja2
     jsonschema
+    markdown
+    markupsafe
+    orjson
+    packaging
+    pep517
+    pip
+    pip-tools
+    python39.pkgs.protobuf
+    pyparsing
+    pyqt6
+    pyqt6-sip
+    pyqt6-webengine
+    pyrsistent
+    pysocks
+    requests
+    send2trash
     setuptools
-  ]
-  ++ lib.optional plotsSupport matplotlib
-  ++ lib.optionals stdenv.isDarwin [ CoreAudio ]
-  ;
+    six
+    soupsieve
+    tomli
+    urllib3
+    waitress
+    werkzeug
+    zipp
+  ] ++ lib.optionals stdenv.isDarwin [ CoreAudio ];
 
-  nativeCheckInputs = [ pytest glibcLocales nose ];
+  # Activate optimizations
+  RELEASE = "1";
 
-  nativeBuildInputs = [ pyqtwebengine.wrapQtAppsHook ];
-  buildInputs = [ lame mpv-unwrapped libpulseaudio ];
+  PROTOC_BINARY = "${protobuf}/bin/protoc";
+  NODE_BINARY = "${nodejs}/bin/node";
 
-  patches = [
-    # Disable updated version check.
-    ./no-version-check.patch
-    (fetchpatch {
-      name = "fix-mpv-args.patch";
```

Comment:
I verified this and the below patches no longer apply. The first is fixed upstream, and the second, I think that code turned into rust and just isn't relevant anymore.